### PR TITLE
[mac_contributing_info] Add instructions for Native Image build on MAC

### PR DESCRIPTION
**Pull request by gitgabrio**
@danielezonca @yesamer @baldimir 

Add instructions for native image build on MAC

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](CONTRIBUTING.md)
- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request checks</b>  
  Please add comment: <b>Jenkins retest this</b>

- for a <b>specific pull request check</b>  
  Please add comment: <b>Jenkins (re)run [kogito-runtimes|kogito-apps|kogito-examples] tests</b>

- for <b>quarkus branch checks</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins run quarkus-branch</b>

- for a <b>quarkus branch specific check</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins (re)run [kogito-runtimes|kogito-apps|kogito-examples] quarkus-branch</b>

- for <b>quarkus main checks</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins run quarkus-main</b>

- for a <b>specific quarkus main check</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins (re)run [kogito-runtimes|kogito-apps|kogito-examples] quarkus-main</b>

- for <b>quarkus lts checks</b>  
  Run checks against Quarkus lts branch  
  Please add comment: <b>Jenkins run quarkus-lts</b>

- for a <b>specific quarkus lts check</b>  
  Run checks against Quarkus lts branch  
  Please add comment: <b>Jenkins (re)run [kogito-runtimes|kogito-apps|kogito-examples] quarkus-lts</b>

- for <b>native checks</b>  
  Run native checks  
  Please add comment: <b>Jenkins run native</b>

- for a <b>specific native check</b>  
  Run native checks 
  Please add comment: <b>Jenkins (re)run [kogito-runtimes|kogito-apps|kogito-examples] native</b>

- for <b>mandrel checks</b>  
  Run native checks against Mandrel image
  Please add comment: <b>Jenkins run mandrel</b>

- for a <b>specific mandrel check</b>  
  Run native checks against Mandrel image  
  Please add comment: <b>Jenkins (re)run [kogito-runtimes|kogito-apps|kogito-examples] mandrel</b>

- for <b>mandrel lts checks</b>  
  Run native checks against Mandrel image and quarkus lts branch
  Please add comment: <b>Jenkins run mandrel-lts</b>

- for a <b>specific mandrel lts check</b>  
  Run native checks against Mandrel image and quarkus lts branch
  Please add comment: <b>Jenkins (re)run [kogito-runtimes|kogito-apps|kogito-examples] mandrel-lts</b>
 
- <b>Full Kogito testing</b> (with cloud images and operator BDD testing)  
  Please add comment: <b>Jenkins run BDD</b>  
  <b>This check should be used only if a big change is done as it takes time to run, need resources and one full BDD tests check can be done at a time ...</b>
</details>

### CODE REVIEW

File: Develop_on_Mac.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Setup MAC for Native image build
+================================
+
+1. install sdkman: curl -s "https://get.sdkman.io" | bash
```
Comment (danielezonca) [inline Develop_on_Mac.md:4]: I would clarify that this step is optional

File: Develop_on_Mac.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Setup MAC for Native image build
+================================
+
+1. install sdkman: curl -s "https://get.sdkman.io" | bash 
+2. install brew: /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+3. manually install Graal 22.3-11: brew install --cask graalvm/tap/graalvm-ce-java11 
+4. link Graal inside sdk: sdk install java 22.3.r11-grl /Library/Java/JavaVirtualMachines/graalvm-ce-java11-22.3.0/Contents/Home 
+5. cd kogito-apps/jitexecutor
```
Comment (danielezonca) [inline Develop_on_Mac.md:8]: Leftover

File: Develop_on_Mac.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+Setup MAC for Native image build
+================================
+
+1. install sdkman: curl -s "https://get.sdkman.io" | bash 
+2. install brew: /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+3. manually install Graal 22.3-11: brew install --cask graalvm/tap/graalvm-ce-java11 
+4. link Graal inside sdk: sdk install java 22.3.r11-grl /Library/Java/JavaVirtualMachines/graalvm-ce-java11-22.3.0/Contents/Home
```
Comment (danielezonca) [inline Develop_on_Mac.md:7]: These "fixed" versions are going to be tricky to maintain (aka remember to update when we maybe move to GraalVM 23 or another version. I would suggest you to make this explicit
```suggestion
3. manually install GraalVM 22.3-11 (Note: check [Quarkus documentation](https://quarkus.io/guides/building-native-image#configuring-graalvm) to get current supported version): brew install --cask graalvm/tap/graalvm-ce-java11 
4. link Graal inside sdk: sdk install java 22.3.r11-grl /Library/Java/JavaVirtualMachines/graalvm-ce-java11-22.3.0/Contents/Home 
```

File: Develop_on_Mac.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+Setup MAC for Native image build
+================================
+
+1. install sdkman: curl -s "https://get.sdkman.io" | bash 
+2. install brew: /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+3. manually install Graal 22.3-11: brew install --cask graalvm/tap/graalvm-ce-java11
```
Comment (danielezonca) [inline Develop_on_Mac.md:6]:
```suggestion
3. manually install GraalVM 22.3-11: brew install --cask graalvm/tap/graalvm-ce-java11 
```

File: Develop_on_Mac.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
+Setup MAC for Native image build
+================================
+
+1. install sdkman: curl -s "https://get.sdkman.io" | bash 
+2. install brew: /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+3. manually install Graal 22.3-11: brew install --cask graalvm/tap/graalvm-ce-java11 
+4. link Graal inside sdk: sdk install java 22.3.r11-grl /Library/Java/JavaVirtualMachines/graalvm-ce-java11-22.3.0/Contents/Home 
+5. cd kogito-apps/jitexecutor 
+6. use Graal vm: sdk use java 22.3.r11-grl
```
Comment (danielezonca) [inline Develop_on_Mac.md:9]:
```suggestion
6. use GraalVM: sdk use java 22.3.r11-grl
```

File: Develop_on_Mac.md (hunk 1 above)
Comment (gitgabrio) [reply Develop_on_Mac.md:4]: done

File: Develop_on_Mac.md (hunk 4 above)
Comment (gitgabrio) [reply Develop_on_Mac.md:6]: done

File: Develop_on_Mac.md (hunk 3 above)
Comment (gitgabrio) [reply Develop_on_Mac.md]: done

File: Develop_on_Mac.md (hunk 2 above)
Comment (gitgabrio) [reply Develop_on_Mac.md]: done

File: Develop_on_Mac.md (hunk 5 above)
Comment (gitgabrio) [reply Develop_on_Mac.md]: done

**Review by danielezonca — APPROVED** [review]

**Review by yesamer — APPROVED** [review]

**Pull request merged**